### PR TITLE
Update hesa reference data for 2021 22 cycle

### PR DESCRIPTION
**Pull request by steventux**
## Context

HESA releases a data collection for each recruitment cycle. We use this to collect candidate information eg. qualifications.
We need to update some values in line with 2021/2022 HESA entries.

## Changes proposed in this pull request

Adds 2 values for `DEGSBJ` in `HESA_DEGREE_SUBJECTS` initializer.
Adds 135 values for `DEGEST` in `HESA_DEGREE_INSTITUTIONS` initializer.

## Guidance to review

I've made the assumption that because the initializers are used to build lists for the candidate forms, given they relate to past qualifications it should be safe to append new values rather than attempting to present year-specific lists.

`DEGCLSS` discrepancies are intentional - we don't differentiate between a _pass_ and a _pass with honours_. See https://github.com/DFE-Digital/apply-for-teacher-training/pull/2240/commits/e5bf63068f435d5329bf5a3cf0a0847d0a917542 

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by malcolmbaig — APPROVED** [review]
lgtm

**Comment by edwardhorsford** [comment]
A heads up that now Register is importing applications from Apply, we sort of need these values too or things can break. Would be great if we could share common lists.

**Comment by frankieroberto** [comment]
Seems like a good idea to add these, but it would be nice to have a better source for the institution names - some of these names seem like they wouldn't necessarily be the ones that graduates recognise – eg "BCNO Limited" is "British College of Osteopathic Medicine". At the minimum we could probably strip "Ltd" and "Limited" from the names?

**Comment by frankieroberto** [comment]
Hmm, digging into this a bit more, I think a lot of the additional entries are institutions that don't have the ability to award degrees, so probably shouldn't be in our list – particularly as they might show up in the autocomplete above more likely options.

**Comment by cwrw** [comment]
I think there is a risk with leaving those out as well as some do award degrees, even if a small subset. If we don't have everything and someone selects one of those, we will not map it correctly to a HESA code and send missing data to institutions. I would go with adding those all now, and maybe we prune the list in the future if it proved problematic

File: config/initializers/qualifications/hesa_degree_subjects.rb [hunk 1]
```diff
@@ -1092,5 +1092,7 @@
   ['101509', 'Fascism'],
   ['101510', 'Environmentalism'],
   ['101511', 'Ophthalmic dispensing'],
+  ['999998', 'Not applicable'],
+  ['999999', 'Not known'],
```
Comment (frankieroberto) [inline config/initializers/qualifications/hesa_degree_subjects.rb:1096]: I don't think these are intended to be user-facing...?

```suggestion
```

**Comment by frankieroberto** [comment]
> I think there is a risk with leaving those out as well as some do award degrees, even if a small subset. If we don't have everything and someone selects one of those, we will not map it correctly to a HESA code and send missing data to institutions. I would go with adding those all now, and maybe we prune the list in the future if it proved problematic

I think we should hold off for now - at least until we can source a more user-facing term for those institutions (eg their trading name rather than company names)?

Candidates can already enter institutions as free text, so presumably providers have an existing process for reporting these.

I’m currently looking at improving the usability of the degree autocompletes, which might let us prioritise the more common institutions, and then extending the list to cover alternative institutions which don't award degree might be less problematic.

**Comment by edwardhorsford** [comment]
I think we should hold off temporarily - and coordinate between our two teams. Register has recently done analysis to compare Apply / Register / DTTP / GOV.UK / Office for students. We should look to add legitimate institutions and map those to HESA - but adding random values reduces the quality and usability of the autocompletes.

From the HESA page:
> Valid entries have been updated to include every provider that is submitting to the Student or Student Alternative records.

I think that means they're now including non-degree institutions in the degree institution list - which means we may need to reconsider how this dataset is ued.

**Comment by cwrw** [comment]
On slack [thread](https://ukgovernmentdfe.slack.com/archives/CAHBLU6ET/p1633347171100700)  but for visibility: we're going to have a meeting in person and decide what we'd like to do next regarding holding off those changes vs implementing them

**Review by cwrw — APPROVED** [review]

**Pull request merged**